Wireshark: strict compilation fails after recent change

The change made in #132 causes a build failure for me. Using an in-tree build based on Wireshark's `release-4.2` branch (the development trunk for 4.2.x builds), compiling with clang-15 on Ubuntu 22.04 jammy.

Apparently the change I made to `timediff()` in #128 isn't as universal as I'd hoped, or else you wouldn't have needed to make #132. Cross-platform portable source code is still a hard problem to solve, evidently. The full error message is:

     /.../wireshark/plugins/epan/ja4/packet-ja4.c:672:51: error: format specifies type 'long' but the argument has type 'long long' [-Werror,-Wformat]
                        wmem_strbuf_append_printf(display, "%" PRId64, (long long) diff);
                                                            ~~~~~~~~~  ^~~~~~~~~~~~~~~~
    /.../wireshark/plugins/epan/ja4/packet-ja4.c:679:53: error: format specifies type 'long' but the argument has type 'long long' [-Werror,-Wformat]
                        wmem_strbuf_append_printf(display, "-R%" PRId64, (long long) diff);
                                                              ~~~~~~~~~  ^~~~~~~~~~~~~~~~
    2 errors generated.

One possible solution would be to change `timediff()` to return an `int64_t`. `PRId64` is meant to hold the specific printf format string needed to print a `int64_t`, and  both come from `<inttypes.h>` (which is included via `<wireshark.h>`).

It might make more logical sense for `timediff()` to return a `time_t`, but it's another one of those types that has different underlying representations on different systems. If there's a portable printf format string for `time_t` I'm not aware of it.

--- a/wireshark/source/packet-ja4.c
+++ b/wireshark/source/packet-ja4.c
@@ -243,18 +243,11 @@ wmem_map_t *conn_hash = NULL; // = wmem_map_new(wmem_file_scope(), g_direct_hash
 wmem_map_t *quic_conn_hash = NULL; // Added for JA4L on quic
 wmem_map_t *packet_table = NULL;
 
-static long timediff(nstime_t *current, nstime_t *prev)
+static int64_t timediff(nstime_t *current, nstime_t *prev)
 {
     nstime_t result;
-    result.secs  = current->secs  - prev->secs;
-    result.nsecs = current->nsecs - prev->nsecs;
-    if (result.nsecs < 0) {
-        --result.secs;
-        result.nsecs += 1000000000L;
-    }
-    float nsecs = ((float)result.nsecs / 1000000000);
-    long diff = result.secs + (((nsecs - floor(nsecs))> 0.5) ? 1 : 0);
-    return diff;
+    nstime_delta(&result, current, prev);
+    return (int64_t)(round(nstime_to_sec(&result)));
 }
 
 pkt_info_t *packet_table_lookup (int frame_number) {
@@ -673,15 +666,15 @@ char *ja4t (ja4t_info_t *data, conn_info_t *conn) {
 	if ((conn != NULL) && (conn->syn_ack_count > 1)) {
 		wmem_strbuf_append_printf(display, "%c", '_');
 		for (int i=1; i<conn->syn_ack_count; i++) {
-			long diff = timediff(&conn->syn_ack_times[i], &conn->syn_ack_times[i-1]);
-			wmem_strbuf_append_printf(display, "%" PRId64, (long long) diff); 
+			int64_t diff = timediff(&conn->syn_ack_times[i], &conn->syn_ack_times[i-1]);
+			wmem_strbuf_append_printf(display, "%" PRId64, diff);
 			if (i < (conn->syn_ack_count - 1)) {
 				wmem_strbuf_append_printf(display, "%c", '-');
 			}
 		}
 		if (!nstime_is_zero(&conn->rst_time)) {
-			long diff = timediff(&conn->rst_time, &conn->syn_ack_times[conn->syn_ack_count-1]);
-			wmem_strbuf_append_printf(display, "-R%" PRId64, (long long) diff); 
+			int64_t diff = timediff(&conn->rst_time, &conn->syn_ack_times[conn->syn_ack_count-1]);
+			wmem_strbuf_append_printf(display, "-R%" PRId64, diff);
 		}
 	}
 

